spec/lib/sidekiq/arguments_filter_spec.rb

```diff
@@ -0,0 +1,22 @@
+RSpec.describe Sidekiq::ArgumentsFilter do
+  describe ".filter_arguments!" do
+    let(:job_args) do
+      [
+        { "id" => 22, "email" => "user@example.com", "name" => "John Doe", "timestamp" => "2024-10-07T12:34:56Z" },
+        { "organisation_id" => 23, "password" => "secret", "created_at" => "2024-10-07T12:34:56Z" },
```

Comment:
Quid des arguments qui sont des hash dans des hash ? Est ce qu'on itére dessus pour le filtrage ? Ex les agents :
```{"id"=>XX, "address"=>"4 rue de la paix, Paris", "agents"=>[{"id"=>XX, "email"=>"email@agent1.fr", "first_name"=>"Agent", "inclusion_connect_open_id_sub"=>nil, "last_name"=>"1"}, {"id"=>XX, "email"=>"email@agent2.fr", "first_name"=>"agent", "inclusion_connect_open_id_sub"=>nil, "last_name"=>"2"}],..., {"model"=>"Rdv", "event"=>"updated", "webhook_reason"=>nil, "timestamp"=>"2024-10-17 11:44:41 +0200"} ```

Reply:
Oui carrément, `ActiveSupport::ParameterFilter` gère les nested hash:  https://api.rubyonrails.org/v7.2/classes/ActiveSupport/ParameterFilter.html